File: ActivityPlanning-Backend/activity_service/src/main/java/dev/activity/entities/Activity.java
package dev.activity.entities;

import java.time.LocalDate;
import java.time.LocalTime;
import java.util.UUID;

import dev.constants.Category;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "activity")
public class Activity {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(nullable = false, length = 30)
    private String activityName;

    @Column(nullable = false, length = 200)
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Category category;

    @Column(nullable = false)
    private LocalDate activityStartingDate;

    @Column(nullable = false)
    private LocalTime activityStartingTime;

    @Column
    private Long activityDurationEstimate;

    @Column(nullable = false)
    private Double latitude;

    @Column(nullable = false)
    private Double longitude;

}

File: ActivityPlanningSystem-Frontend/ActivityPlanningSystem-Frontend/Model/Activity.swift
//
//  Activity.swift
//  ActivityPlanningSystem-Frontend
//
//  Created by Ege Kara on 20.02.2025.
//

import Foundation

struct Activity : Codable , Identifiable {
    
    let id : UUID
    let activityName : String
    let description : String
    let category : String
    let activityStartingDate : String
    let activityStartingTime : String
    let activityDurationEstimate : Int32
    let latitude : Double
    let longitude : Double
    
    init(id: UUID, activityName: String, description: String, category: String, activityStartingDate: String, activityStartingTime: String, activityDurationEstimate: Int32, latitude: Double, longitude: Double) {
        self.id = id
        self.activityName = activityName
        self.description = description
        self.category = category
        self.activityStartingDate = activityStartingDate
        self.activityStartingTime = activityStartingTime
        self.activityDurationEstimate = activityDurationEstimate
        self.latitude = latitude
        self.longitude = longitude
    }
}

struct ActivityRequest : Codable {
    
    private let activityName : String
    private let description : String
    private let category : String
    private let activityStartingDate : String
    private let activityStartingTime : String
    private let activityDurationEstimate : Int32
    private let latitude : Double
    private let longitude : Double
    
    init(activityName: String, description: String, category: String, activityStartingDate: String, activityStartingTime: String, activityDurationEstimate: Int32, latitude: Double, longitude: Double) {
        self.activityName = activityName
        self.description = description
        self.category = category
        self.activityStartingDate = activityStartingDate
        self.activityStartingTime = activityStartingTime
        self.activityDurationEstimate = activityDurationEstimate
        self.latitude = latitude
        self.longitude = longitude
    }
    
}

struct MarkModel : Codable {
    
    private let id : String
    private let latitude : Double
    private let longitude : Double
    
    init(id: String, lattitude: Double, longitude: Double) {
        self.id = id
        self.latitude = lattitude
        self.longitude = longitude
    }
    
}
